src/test/java/pageObjects/FilterTest.java

```diff
@@ -0,0 +1,65 @@
+package pageObjects;
+
+import org.openqa.selenium.By;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import utils.Constants;
+
+public class FilterTest extends BaseTest {
+    @Test
+    public void filterBySubcategory() {
+        filter.filterBySubcategory();
+        String table = driver.findElement(By.id("device-table")).getText();
+        Assert.assertTrue(table.contains("LAPTOP"));
+    }
+    @Test
+    public void filterByName() {
+        filter.filterByName();
+        String table = driver.findElement(By.id("device-table")).getText();
+        Assert.assertTrue(table.contains("Name"));
+    }
```

Comment:
correct me if im wrong but if we get the text from the whole table, we also get the header which has a column called "Name". this means we will always have this test pass even if the filter is not working as long as the device table loads